src/interface/misc.rs

```diff
@@ -11,6 +11,10 @@ pub use parking_lot::{
     Condvar, Mutex, RwLock as RustLock, RwLockReadGuard as RustLockReadGuard,
     RwLockWriteGuard as RustLockWriteGuard,
 };
+use std::slice;
+use crate::safeposix::cage::syscall_error;
```

Comment:
id rather we not import the errno stuff in here, can you have it return an error and handle the errno in the syscall